homeassistant/components/fibaro/sensor.py

```diff
@@ -114,3 +132,52 @@ def update(self):
         """Update the state."""
         with suppress(KeyError, ValueError):
             self.current_value = float(self.fibaro_device.properties.value)
+
+
+class FibaroEnergySensor(FibaroDevice, SensorEntity):
```

Comment:
Shouldn't the energy attributes from the parent class be removed?

Reply:
That is what I thought as well.
- Then it is a breaking change
- Also I do not know where the current energy in watt should be, in the energy sensor or in the original entity?
- Also there are separate entities for current and voltage, should they as well be grouped in another way? The fibaro devices have quite a different structure than we have in HA, so it is quite difficult to change here something.

Reply:
> Then it is a breaking change

Yes :)

> Also I do not know where the current energy in watt should be

Energy in Watt? You are mixing up power & energy now 😉 

Anyways, so, the `current_power_w` and `today_energy_kwh` from `switch.py` are deprecated and both should be extracted into separate new sensors (and removed from `switch.py`)

> Also there are separate entities for current and voltage, should they as well be grouped in another way?

Each data point (including current and voltage) should become its own entity.
Entities can be grouped into devices, which this integration currently doesn't seem to implement?

Reply:
Ok I removed the attribute current_energy_kwh from other entities. Also I fixed two other things which have been wrong, namely I set now a correct unique_id for the energy entities. Also I changed the callback mechanism as now more than one fibaro entity can have the same id (this id is the id given by fibaro to a fibaro device).